library/log/src/modules/LogViewCore.rb

```diff
@@ -120,19 +121,20 @@ def Start(widget, d)
 
       if command == ""
         if grep == ""
-          command = Builtins.sformat("tail -n %1 -f '%2'", max_lines, file)
+          command = Builtins.sformat("/usr/bin/tail -n %1 -f %2",
+            max_lines.to_s.shellescape, file.shellescape)
         else
           command = Builtins.sformat(
-            "tail -n +0 -f '%1' | grep --line-buffered '%2'",
-            file,
-            grep
+            "tail -n +0 -f %1 | grep --line-buffered %2",
+            file.shellescape,
+            grep.shellescape
           )
 
           if max_lines != 0
             lc_command = Builtins.sformat(
-              "cat '%1' | grep '%2' | wc -l",
-              file,
-              grep
+              "/usr/bin/cat %1 | /usr/bin/grep %2 | /usr/bin/wc -l",
```

Comment:
Simply use `/usr/bin/grep -c` instad of the `grep | wc -l` pipeline?

Reply:
...even worse, that `cat` is completely redundant. Simply use

`/usr/bin/grep -c %2 <%1`

Reply:
even stdin redirect is not needed and simple `/usr/bin/grep -c %2 %1` is enough

Reply:
You are right. So let's simplify this to get it better readable and maintainable.